solr/core/src/java/org/apache/solr/api/V2HttpCall.java

```diff
@@ -360,6 +366,85 @@ protected void execute(SolrQueryResponse rsp) {
     SolrCore.postDecorateResponse(handler, solrReq, rsp);
   }
 
+  @Override
+  protected void populateTracingSpan(Span span) {
+    // Set db.instance
+    String coreOrColName = this.origCorename;
+    if (coreOrColName == null) {
+      Map<String, String> pathTemplateValues = getUrlParts(); // == solrReq.getPathTemplateValues()
+      coreOrColName = pathTemplateValues.get("collection");
+      if (coreOrColName == null) {
+        coreOrColName = pathTemplateValues.get("core");
+      }
+    }
+    if (coreOrColName != null) {
+      span.setTag(Tags.DB_INSTANCE, coreOrColName);
+    }
+
+    // Get the templatize-ed path
+    String path;
+    if (api instanceof AnnotatedApi) {
+      // ideal scenario; eventually everything might be AnnotatedApi?
+      var aapi = (AnnotatedApi) api;
+      path = aapi.getEndPoint().path()[0]; // consider first to be primary
+    } else {
+      path = computeEndpointPath();
+      // TODO consider getValidators, looking for command & path?
+    }
+
+    String verb = req.getMethod().toLowerCase(Locale.ROOT);
+    // if this api has commands ...
+    final Map<String, JsonSchemaValidator> validators = getValidators(); // should be cached
+    if (validators != null && validators.isEmpty() == false && solrReq != null) {
+      // does this request have one command?
+      List<CommandOperation> cmds = solrReq.getCommands(false);
+      if (cmds.size() == 1) {
+        verb = cmds.get(0).name;
+      }
+    }
+
+    span.setOperationName(verb + ":" + path);
+
+    if (req.getQueryString() != null) {
+      span.setTag("params", req.getQueryString());
```

Comment:
would "params" need to be added as a constant? for example `Tags.PARAMS`

Reply:
I don't think there is value in a constant for this use-case (basically a form of logging).  If hypothetically this was parsed within the codebase then it would make sense to me as it would also link the dependency between read/write.

Also, I think this should be namespaced... like "http.params".  The names seem somewhat loose; "http." is not reserved, so I think it's okay to add it there.

Reply:
In a commit soon I moved this to SolrDispatchFilter as http.params where there is no duplication and also where it's aligned with the "http" tags.